File: spaces/params/page.go
package params

import (
	"github.com/fox-gonic/fox/database"
	"github.com/miclle/space/models"
)

// CreatePage create page params
type CreatePage struct {
	SpaceID    int64
	CreatorID  int64
	ParentID   int64
	Lang       string
	Version    string
	Status     models.PageStatus
	Title      string
	ShortTitle string
	Body       string
}

// DescribePages describe page detail params
type DescribePages struct {
	database.Pagination[*models.Page]
	SpaceID int64
	Lang    string
	Version string
	Depth   string
	View    string // list | tree
}

// DescribePage describe page detail params
type DescribePage struct {
	SpaceID int64
	PageID  int64
	Lang    string
	Version string
}

// UpdatePage update page params
type UpdatePage struct {
	ID         int64
	Lang       *string
	Version    *string
	Status     *models.PageStatus
	Title      *string
	ShortTitle *string
	Body       *string
}

// Search page params
type Search struct {
	database.Pagination[*models.PageContent]
	Lang string
	Q    string
}

File: ui/src/services/page.ts
import { GET, PATCH, POST } from './lib/http';
import { Nullish } from './lib/types';

import { IPage, PageStatus } from 'models';

export interface ICreatePageArgs {
  parent_id?:        number
  page_id?:          number
  title:             string
  version:           string
  status:            PageStatus
  short_title?:      string
  body:              string
}

export function create(spaceKey: string, args: ICreatePageArgs): Promise<IPage> {
  return POST(`/spaces/${spaceKey}/pages`, args)
}
export interface IGetPageParams {
  lang?:    string | Nullish
  version?: string | Nullish
  depth?:   number | Nullish
}

export function list(spaceKey: string, params?: IGetPageParams): Promise<IPage[]> {
  return GET(`/spaces/${spaceKey}/pages`, { params })
}

export function get(spaceKey: string, id: string, params?: IGetPageParams): Promise<IPage> {
  return GET(`/spaces/${spaceKey}/pages/${id}`, { params })
}

export type IUpdatePageArgs = ICreatePageArgs;

export function update(spaceKey: string, id: string, args: IUpdatePageArgs): Promise<IPage> {
  return PATCH(`/spaces/${spaceKey}/pages/${id}`, args)
}
